### ISSUE TITLE
[Routing] Invalid matches, possibly invalid regex dumped

### ISSUE THREAD

**Issue by Pierstoval** [symfony/symfony#31253]
**Symfony version(s) affected**: 4.2.7, does not seem to be present in 4.2.6 (so downgrading can be a temporary solution)

**Description**  
When creating a route based on a host, a E_NOTICE error appears in the `PhpMatcherTrait`.

**How to reproduce**  
Here's an example project: https://github.com/Pierstoval/routing-issue-reproducer

You can use the native Symfony binary to set it up with a host:

```
symfony proxy:domain:attach backend.app
```

Just warm up the cache and check goo to https://backend.app.docker/fr/ and you'll see something like this:

![image](https://user-images.githubusercontent.com/3369266/56757381-57f4ea00-6794-11e9-9e5a-9104995b7a28.png)

You can also check `var/cache/dev/srcApp_KernelDevDebugContainerUrlMatcher.php` and you'll see something like this:

```php
$this->regexpList = [
    0 => '{^(?'
        .'|(?:(?:[^./]*+\\.)++)(?'
            .'|/_(?'
                .'|error/(\\d+)(?:\\.([^/]++))?(*:60)'
                .'|wdt/([^/]++)(*:79)'
                .'|profiler/([^/]++)(?'
                    .'|/(?'
                        .'|search/results(*:124)'
                        .'|router(*:138)'
                        .'|exception(?'
                            .'|(*:158)'
                            .'|\\.css(*:171)'
                        .')'
                    .')'
                    .'|(*:181)'
                .')'
            .')'
        .')|(?i:backend\\.app\\.docker)\\.(?'
            .'|/((?:fr|en))(?:/((?:Factions|Maps|Markers|MarkersTypes|Routes|RoutesTypes|Zones|ZonesTypes|TransportTypes|PortalElement|Subscriptions|Users))(?:/([^/]++)(?:/([^/]++))?)?)?(*:393)'
        .')'
        .')/?$}sDu',
];
```

I'm not sure having the host in the regex is expected.

I hope I gave enough details to reproduce.

Also, I'm on Windows, and using PHP 7.3.4

**Comment by nicolas-grekas** [comment]
We fixed a similar issue in 4.2.8, can you please check again after upgrading?

**Comment by Pierstoval** [comment]
Installed `4.2.8` and it seems fixed 👍
